src/infix-longscale.ts

```diff
@@ -0,0 +1,52 @@
+
+// Copy-pasted from standard.ts
+const ABBREVIATIONS = [
+  ",", "K", "M", ".", "B", ".", "T", ".", "Qa", ".", "Qt", ".", "Sx", ".",
+  "Sp", ".", "Oc", ".", "No", ".", "Dc", ".", "UDc", ".", "DDc",
+  "TDc", ".", "QaDc", ".", "QtDc", ".", "SxDc", ".", "SpDc", ".", "ODc", ".",
+  "NDc", ".", "Vg", ".", "UVg", ".", "DVg", ".", "TVg",
+  "QaVg", ".", "QtVg", ".", "SxVg", ".", "SpVg", ".", "OVg", ".", "NVg", ".",
+  "Tg", ".", "UTg", ".", "DTg", ".", "TTg", ".", "QaTg",
+  "QtTg", ".", "SxTg", ".", "SpTg", ".", "OTg", ".", "NTg", ".", "Qd", ".",
+  "UQd", ".", "DQd", ".", "TQd", ".", "QaQd", ".", "QtQd",
+  "SxQd", ".", "SpQd", ".", "OQd", ".", "NQd", ".", "Qi", ".", "UQi", ".",
+  "DQi", ".", "TQi", ".", "QaQi", ".", "QtQi", ".", "SxQi",
+  "SpQi", ".", "OQi", ".", "NQi", ".", "Se", ".", "USe", ".", "DSe", ".",
+  "TSe", ".", "QaSe", ".", "QtSe", ".", "SxSe", ".", "SpSe",
+  "OSe", ".", "NSe", ".", "St", ".", "USt", ".", "DSt", ".", "TSt", ".",
+  "QaSt", ".", "QtSt", ".", "SxSt", ".", "SpSt", ".", "OSt",
+  "NSt", ".", "Og", ".", "UOg", ".", "DOg", ".", "TOg", ".", "QaOg", ".",
+  "QtOg", ".", "SxOg", ".", "SpOg", ".", "OOg", ".", "NOg",
+  "Nn", ".", "UNn", ".", "DNn", ".", "TNn", ".", "QaNn", ".", "QtNn", ".",
+  "SxNn", ".", "SpNn", ".", "ONn", ".", "NNn", ".", "Ce"
+];
+
+
+import { AbstractInfixNotation } from "./infix-abstract";
+import Decimal from "break_infinity.js";
+
+// Name comes from https://en.wikipedia.org/wiki/Long_and_short_scales
+export class InfixLongScaleNotation extends AbstractInfixNotation {
+  public get name(): string {
+    return "Infix long scale";
+  }
+
+  public formatDecimal(value: Decimal, places: number): string {
+    let str = super.formatDecimal(value, Math.max(places, 5));
+
+    if ((value.exponent > 0) && (value.exponent % 6) === 5) {
+      str += this.formatExponent(value.exponent - 5);
+    }
+    return str;
+  }
+
+  protected formatMantissa(digit: number): string {
+    return this.numberToSubscript(digit);
+  }
+  protected formatExponent(exp: number): string {
+    const i = exp / 3;
+    return (i > -1 && i <= 101)
+      ? ABBREVIATIONS[i]
+      : (i-1).toString(10);
```

Comment:
I'm a bit curious as to why you don't do what standard does here and use more abbreviations. If it's because long scale doesn't do that, that makes sense, but suggests that I (or you if you want) should add long scale as a new notation itself.

Reply:
Because that would be quite a lot of code duplication IMHO. Would you agree to refactor the standard abbreviations away into a separate file, so that they can be reused by the infix notation without code duplication?